Can't create new one ai when user set <appInsightsInstance> in configuration

### Plugin name and version
azure-functions-maven-plugin 1.4.2-SNAPSHOT

### Plugin configuration in your `pom.xml`
```
 <configuration>
 <appName>${functionAppName}</appName>
 <resourceGroup>harrytest12</resourceGroup>
 <appServicePlanName>harrytest12-app-service-plan</appServicePlanName>
 <region>westus</region>
 <runtime>
 <os>windows</os>
 </runtime>
 <appInsightsInstance>harryai12</appInsightsInstance>
 <appSettings>
 <property>
 <name>FUNCTIONS_EXTENSION_VERSION</name>
 <value>~3</value>
 </property>
 </appSettings>
 </configuration>
```

### Actual behavior

![image](https://user-images.githubusercontent.com/41051257/83394596-71527600-a42b-11ea-9868-cc3bfff80643.png)

```
INFO] Auth Type : AZURE_CLI, Auth Files : [/Users/user/.azure/azureProfile.json, /Users/user/.azure/accessTokens.json]
[INFO] Subscription : XXXXXXXXXX
[INFO] The specified function app does not exist. Creating a new function app...
[INFO] Set function worker runtime to java.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 15.219 s
[INFO] Finished at: 2020-06-01T17:07:05+08:00
[INFO] Final Memory: 68M/353M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.microsoft.azure:azure-functions-maven-plugin:1.4.2-SNAPSHOT:deploy (default-cli) on project test01: java.lang.NullPointerException: OnError while emitting onNext value: null -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.microsoft.azure:azure-functions-maven-plugin:1.4.2-SNAPSHOT:deploy (default-cli) on project test01: java.lang.NullPointerException
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:216)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:862)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:286)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:197)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: java.lang.NullPointerException
        at com.microsoft.azure.maven.AbstractAzureMojo.handleException(AbstractAzureMojo.java:509)
        at com.microsoft.azure.maven.AbstractAzureMojo.execute(AbstractAzureMojo.java:443)
        at com.microsoft.azure.maven.function.AbstractFunctionMojo.execute(AbstractFunctionMojo.java:183)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
        ... 20 more
Caused by: java.lang.NullPointerException
        at com.microsoft.azure.management.applicationinsights.v2015_05_01.implementation.ComponentsImpl.wrapModel(ComponentsImpl.java:145)
        at com.microsoft.azure.management.applicationinsights.v2015_05_01.implementation.ComponentsImpl.wrapModel(ComponentsImpl.java:29)
        at com.microsoft.azure.arm.resources.collection.implementation.GroupableResourcesCoreImpl$1.call(GroupableResourcesCoreImpl.java:118)
        at com.microsoft.azure.arm.resources.collection.implementation.GroupableResourcesCoreImpl$1.call(GroupableResourcesCoreImpl.java:115)
        at rx.internal.operators.OnSubscribeMap$MapSubscriber.onNext(OnSubscribeMap.java:69)
        at rx.internal.operators.OnSubscribeMap$MapSubscriber.onNext(OnSubscribeMap.java:77)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.emitScalar(OperatorMerge.java:511)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.tryEmit(OperatorMerge.java:466)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.onNext(OperatorMerge.java:244)
        at rx.internal.operators.OperatorMerge$MergeSubscriber.onNext(OperatorMerge.java:148)
        at rx.internal.operators.OnSubscribeMap$MapSubscriber.onNext(OnSubscribeMap.java:77)
        at retrofit2.adapter.rxjava.CallArbiter.deliverResponse(CallArbiter.java:120)
        at retrofit2.adapter.rxjava.CallArbiter.emitResponse(CallArbiter.java:102)
        at retrofit2.adapter.rxjava.CallExecuteOnSubscribe.call(CallExecuteOnSubscribe.java:46)
        at retrofit2.adapter.rxjava.CallExecuteOnSubscribe.call(CallExecuteOnSubscribe.java:24)
        at rx.Observable.unsafeSubscribe(Observable.java:10327)
        at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:48)
        at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:33)
        at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:48)
        at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
        at rx.Observable.unsafeSubscribe(Observable.java:10327)
        at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:48)
        at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:33)
        at rx.Observable.unsafeSubscribe(Observable.java:10327)
        at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:48)
        at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:33)
        at rx.Observable.unsafeSubscribe(Observable.java:10327)
        at rx.internal.operators.DeferredScalarSubscriber.subscribeTo(DeferredScalarSubscriber.java:153)
        at rx.internal.operators.OnSubscribeTakeLastOne.call(OnSubscribeTakeLastOne.java:32)
        at rx.internal.operators.OnSubscribeTakeLastOne.call(OnSubscribeTakeLastOne.java:22)
        at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:48)
        at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
        at rx.Observable.subscribe(Observable.java:10423)
        at rx.Observable.subscribe(Observable.java:10390)
        at rx.observables.BlockingObservable.blockForSingle(BlockingObservable.java:443)
        at rx.observables.BlockingObservable.last(BlockingObservable.java:226)
        at com.microsoft.azure.arm.resources.collection.implementation.GroupableResourcesCoreImpl.getByResourceGroup(GroupableResourcesCoreImpl.java:110)
        at com.microsoft.azure.arm.resources.collection.implementation.GroupableResourcesCoreImpl.getByResourceGroup(GroupableResourcesCoreImpl.java:34)
        at com.microsoft.azure.common.applicationinsights.ApplicationInsightsManager.getApplicationInsightInstance(ApplicationInsightsManager.java:33)
        at com.microsoft.azure.maven.function.DeployMojo.getApplicationInsights(DeployMojo.java:213)
        at com.microsoft.azure.maven.function.DeployMojo.getOrCreateApplicationInsights(DeployMojo.java:207)
        at com.microsoft.azure.maven.function.DeployMojo.buildApplicationInsightConnection(DeployMojo.java:191)
        at com.microsoft.azure.maven.function.DeployMojo.createFunctionApp(DeployMojo.java:136)
        at com.microsoft.azure.maven.function.DeployMojo.createOrUpdateFunctionApp(DeployMojo.java:126)
        at com.microsoft.azure.maven.function.DeployMojo.doExecute(DeployMojo.java:99)
        at com.microsoft.azure.maven.AbstractAzureMojo.execute(AbstractAzureMojo.java:438)
        ... 23 more
Caused by: rx.exceptions.OnErrorThrowable$OnNextValue: OnError while emitting onNext value: null
        at rx.exceptions.OnErrorThrowable.addValueAsLastCause(OnErrorThrowable.java:118)
        at rx.internal.operators.OnSubscribeMap$MapSubscriber.onNext(OnSubscribeMap.java:73)
        ... 64 more
```

### Steps to reproduce the problem

1. Configure pom.xml as above
2. Run command: mvn clean package
3. Run command: mvn azure-functions:deploy

--- a/azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/DeployMojo.java
+++ b/azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/DeployMojo.java
@@ -190,7 +190,7 @@ public class DeployMojo extends AbstractFunctionMojo {
      * @param isCreation Define the stage of function app, as we only create ai instance by default when create new function apps
      * @throws AzureExecutionException When there are conflicts in configuration or meet errors while finding/creating application insights instance
      */
-    private void bindApplicationInsights(Map appSettings, boolean isCreation) throws AzureExecutionException {
+    private void bindApplicationInsights(Map appSettings, boolean isCreation) throws AzureExecutionException, AzureAuthFailureException {
         // Skip app insights creation when user specify ai connection string in app settings
         if (appSettings.containsKey(APPINSIGHTS_INSTRUMENTATION_KEY)) {
             return;
@@ -213,16 +213,17 @@ public class DeployMojo extends AbstractFunctionMojo {
         }
     }
 
-    private ApplicationInsightsComponent getOrCreateApplicationInsights(boolean enableCreation) throws AzureExecutionException {
+    private ApplicationInsightsComponent getOrCreateApplicationInsights(boolean enableCreation) throws AzureAuthFailureException {
         final AzureTokenCredentials credentials = getAzureTokenWrapper();
         if (credentials == null) {
             Log.warn(APPLICATION_INSIGHTS_NOT_SUPPORTED);
             return null;
         }
-        final ApplicationInsightsManager applicationInsightsManager = new ApplicationInsightsManager(credentials, getUserAgent());
+        final String subscriptionId = getAzureClient().subscriptionId();
+        final ApplicationInsightsManager applicationInsightsManager = new ApplicationInsightsManager(credentials, subscriptionId, getUserAgent());
         return StringUtils.isNotEmpty(getAppInsightsInstance()) ?
                 getApplicationInsights(applicationInsightsManager, getAppInsightsInstance()) :
-                enableCreation ? createApplicationInsights(applicationInsightsManager) : null;
+                enableCreation ? createApplicationInsights(applicationInsightsManager, getAppName()) : null;
     }
 
     private ApplicationInsightsComponent getApplicationInsights(ApplicationInsightsManager applicationInsightsManager,
@@ -231,19 +232,19 @@ public class DeployMojo extends AbstractFunctionMojo {
                 appInsightsInstance);
         if (resource == null) {
             Log.warn(String.format(FAILED_TO_GET_APPLICATION_INSIGHTS, appInsightsInstance, getResourceGroup()));
-            return createApplicationInsights(applicationInsightsManager);
+            return createApplicationInsights(applicationInsightsManager, appInsightsInstance);
         }
         return resource;
     }
 
-    private ApplicationInsightsComponent createApplicationInsights(ApplicationInsightsManager applicationInsightsManager) {
+    private ApplicationInsightsComponent createApplicationInsights(ApplicationInsightsManager applicationInsightsManager, String name) {
         if (isDisableAppInsights()) {
             Log.info(SKIP_CREATING_APPLICATION_INSIGHTS);
             return null;
         }
         try {
             Log.info(APPLICATION_INSIGHTS_CREATE_START);
-            final ApplicationInsightsComponent resource = applicationInsightsManager.createApplicationInsights(getResourceGroup(), getAppName(), getRegion());
+            final ApplicationInsightsComponent resource = applicationInsightsManager.createApplicationInsights(getResourceGroup(), name, getRegion());
             Log.info(String.format(APPLICATION_INSIGHTS_CREATED, resource.name(), resource.id()));
             return resource;
         } catch (Exception e) {

--- a/azure-tools-common/src/main/java/com/microsoft/azure/common/applicationinsights/ApplicationInsightsManager.java
+++ b/azure-tools-common/src/main/java/com/microsoft/azure/common/applicationinsights/ApplicationInsightsManager.java
@@ -6,31 +6,33 @@
 
 package com.microsoft.azure.common.applicationinsights;
 
-import com.microsoft.azure.common.exceptions.AzureExecutionException;
 import com.microsoft.azure.credentials.AzureTokenCredentials;
 import com.microsoft.azure.management.Azure;
 import com.microsoft.azure.management.applicationinsights.v2015_05_01.ApplicationInsightsComponent;
 import com.microsoft.azure.management.applicationinsights.v2015_05_01.ApplicationType;
 import com.microsoft.azure.management.applicationinsights.v2015_05_01.implementation.InsightsManager;
 
-import java.io.IOException;
-
 public class ApplicationInsightsManager {
 
     private Azure azure;
     private InsightsManager insightsManager;
 
-    public ApplicationInsightsManager(AzureTokenCredentials tokenCredentials, String userAgent) throws AzureExecutionException {
-        try {
-            insightsManager = InsightsManager.authenticate(tokenCredentials, tokenCredentials.defaultSubscriptionId());
-            azure = Azure.configure().authenticate(tokenCredentials).withDefaultSubscription();
-        } catch (IOException e) {
-            throw new AzureExecutionException(e.getMessage(), e);
-        }
+    public ApplicationInsightsManager(AzureTokenCredentials tokenCredentials, String subscriptionId, String userAgent) {
+        azure = Azure.configure()
+                .withUserAgent(userAgent)
+                .authenticate(tokenCredentials).withSubscription(subscriptionId);
+        insightsManager = InsightsManager.configure()
+                .withUserAgent(userAgent)
+                .authenticate(tokenCredentials, subscriptionId);
     }
 
     public ApplicationInsightsComponent getApplicationInsightsInstance(String resourceGroup, String name) {
-        return insightsManager.components().getByResourceGroup(resourceGroup, name);
+        try {
+            return insightsManager.components().getByResourceGroup(resourceGroup, name);
+        } catch (Exception e) {
+            // SDK will throw exception when resource not found
+            return null;
+        }
     }
 
     public ApplicationInsightsComponent createApplicationInsights(String resourceGroup, String name, String location) {

